lib/Runtime/Base/ScriptContext.cpp

```diff
@@ -654,6 +654,22 @@ namespace Js
 
                 if (hasFunctions)
                 {
+                    struct AutoReset
+                    {
+                        AutoReset(ThreadContext* threadContext)
+                            :threadContext(threadContext)
+                        {
+                            // indicate background thread that we need help to delete the xData
+                            threadContext->IndicateExtraWork();
+                        }
+                        ~AutoReset()
+                        {
+                            threadContext->IndicateNoMoreExtraWork();
+                        }
+
+                        ThreadContext* threadContext;
+                    } autoReset(this->GetThreadContext());
```

Comment:
Get the jobprocessor here? instead of thread context to void two extra API on thread context?